drivers/ieee802154/ieee802154_mcr20a.c

```diff
@@ -202,12 +202,13 @@ bool _mcr20a_write_burst(struct mcr20a_spi *spi, bool dreg, u16_t addr,
 {
 	bool retval;
 
-	k_sem_take(&spi->spi_sem, K_FOREVER);
```

Comment:
This looks like unrelated change, perhaps this could be placed in another commit.

Reply:
It is related. The coverity bug is a memory overwrite bug. The "off by one" error is fixed by the changing the test but the SYS_LOG_ERR message doesn't prevent the overwrite from happening. Putting a return 0 does prevent the overwrite but requires changing how you deal with the semaphore. I had two choices, 1) either give back the semaphore before returning 0 or 2) moving the "take" to after the size test. I selected option 2.

Reply:
The need for the cmd_buf is in the worst case (dreg is false) len + 2 (one octet for indirect write cmd + one octet for the address). I do not see the issue here, maybe someone can explain it to me.

Reply:
This took me a moment to see as well. Coverity is smart enough to know that the test is checking to see that (len+2) > sizeof(cmd_buf) is potentially problematic later in the code because a maximum value of len at this test could be 10. But later in the code here a value of 10 would have a problem of a one byte over write in:

memcpy(&spi->cmd_buf[2], data_buf, len);